ZuluContent/Spells/First/ReactiveArmor.cs

```diff
@@ -33,26 +36,15 @@ public async Task OnTargetAsync(ITargetResponse<Mobile> response)
             Caster.FireHook(h => h.OnModifyWithMagicEfficiency(Caster, ref charges));
             Caster.FireHook(h => h.OnModifyWithMagicEfficiency(Caster, ref duration));
 
-            if (Caster is IEnchanted enchanted)
+            buffable.BuffManager.AddBuff(new ReactiveArmor
             {
-                enchanted.Enchantments.Set((ReactiveArmor e) => e.Value = (int) charges);
-            }
-
-            Caster.FixedParticles(0x376A, 9, 32, 5008, EffectLayer.Waist);
-            Caster.PlaySound(0x1F2);
-
-            EndActionAsync(mobile, duration);
-        }
-
-        private static async void EndActionAsync(Mobile mobile, double duration)
-        {
-            await Timer.Pause(TimeSpan.FromSeconds(duration));
+                Value = (int)charges,
+                Start = DateTime.UtcNow,
```

Comment:
Just wondering if its needed to specify Start in this? Wouldn't it always be DateTime.UtcNow or is there a case where a buff would start in the future?

Reply:
Yeah in this instance it can be optionally left off as `ReactiveArmor` has a default init for `Start`.

`Start` and `Duration` are mostly remnants the RunUO/ServUO BuffIcon implementation I based the BuffManager on. However I think the potential use-case would be where you create/prepare a buff but only activate it later after some delay, or on some condition. Right now though it only gets set during init.

Reply:
ok